test/lib/jdk/test/lib/containers/docker/DockerTestUtils.java

```diff
@@ -120,59 +120,64 @@ private static boolean isDockerEngineAvailableCheck() throws Exception {
+     * @param dockerfileContent content of the Dockerfile; use null to generate default content
      * @throws Exception
      */
-    public static void
-        buildJdkDockerImage(String imageName, String dockerfile, String buildDirName)
-            throws Exception {
+    public static void buildJdkContainerImage(String imageName, String dockerfileContent) throws Exception {
 
-        Path buildDir = Paths.get(".", buildDirName);
+        // Create an image build/staging directory
+        Path buildDir = Paths.get(".", "image-build");
         if (Files.exists(buildDir)) {
             throw new RuntimeException("The docker build directory already exists: " + buildDir);
         }
+        Files.createDirectories(buildDir);
 
-        Path jdkSrcDir = Paths.get(JDK_UNDER_TEST);
-        Path jdkDstDir = buildDir.resolve("jdk");
-
-        Files.createDirectories(jdkDstDir);
+        // Generate Dockerfile
+        if (dockerfileContent != null) {
+            Files.writeString(buildDir.resolve("Dockerfile"), dockerfileContent);
+        } else {
+            generateDockerFile(buildDir.resolve("Dockerfile"),
+                           DockerfileConfig.getBaseImageName(),
+                           DockerfileConfig.getBaseImageVersion());
+        }
 
         // Copy JDK-under-test tree to the docker build directory.
         // This step is required for building a docker image.
+        Path jdkSrcDir = Paths.get(JDK_UNDER_TEST);
+        Path jdkDstDir = buildDir.resolve("jdk");
+        Files.createDirectories(jdkDstDir);
         Files.walkFileTree(jdkSrcDir, new CopyFileVisitor(jdkSrcDir, jdkDstDir));
-        buildDockerImage(imageName, Paths.get(Utils.TEST_SRC, dockerfile), buildDir);
+
+        buildImage(imageName, buildDir);
     }
 
 
     /**
-     * Build a docker image based on given docker file and docker build directory.
+     * Build a container image based on given Dockerfile and image build directory.
      *
      * @param imageName  name of the image to be created, including version tag
-     * @param dockerfile  path to the Dockerfile to be used for building the docker
-     *        image. The specified dockerfile will be copied to the docker build
-     *        directory as 'Dockerfile'
-     * @param buildDir  build directory; it should already contain all the content
-     *        needed to build the docker image.
+     * @param buildDir   build directory; it should already contain all the content
+     *                   needed to build the image.
      * @throws Exception
      */
-    public static void
-        buildDockerImage(String imageName, Path dockerfile, Path buildDir) throws Exception {
-
-        generateDockerFile(buildDir.resolve("Dockerfile"),
-                           DockerfileConfig.getBaseImageName(),
-                           DockerfileConfig.getBaseImageVersion());
+    public static void buildImage(String imageName, Path buildDir) throws Exception {
```

Comment:
it seems there are no users of this method, do we need to keep it public?

Reply:
Sounds good. Will do.